openmc/model/triso.py

```diff
@@ -1016,81 +1370,57 @@ def pack_trisos(radius, fill, domain_shape='cylinder', domain_length=None,
+
+    # Determine the packing fraction/number of spheres
+    volume = 4/3*pi*radius**3
+    if pf is None and num_spheres is None:
+        raise ValueError('`pf` or `num_spheres` must be specified.')
+    elif pf is None:
+        num_spheres = int(num_spheres)
+        pf = volume*num_spheres/domain.volume
+    else:
+        pf = float(pf)
+        num_spheres = int(pf*domain.volume//volume)
+
+    # Make sure initial packing fraction is less than packing fraction
+    if initial_pf > pf:
+        initial_pf = pf
+
+    # Check packing fraction for close random packing
+    if pf > MAX_PF_CRP:
+        raise ValueError('Packing fraction {0} is greater than the limit for '
+                         'close random packing, {1}'.format(pf, MAX_PF_CRP))
+
+    # Check packing fraction for random sequential packing
+    if initial_pf > MAX_PF_RSP:
+        raise ValueError('Initial packing fraction {0} is greater than the '
+                         'limit for random sequential packing, '
+                         '{1}'.format(initial_pf, MAX_PF_RSP))
+
+    # Calculate the sphere radius used in the initial random sequential
     # packing from the initial packing fraction
-    initial_radius = (3/4 * initial_packing_fraction * domain.volume /
-                      (pi * n_particles))**(1/3)
-    domain.particle_radius = initial_radius
+    initial_radius = (3/4*initial_pf*domain.volume/(pi*num_spheres))**(1/3)
+    domain.sphere_radius = initial_radius
 
     # Recalculate the limits for the initial random sequential packing using
-    # the desired final particle radius to ensure particles are fully contained
+    # the desired final sphere radius to ensure spheres are fully contained
     # within the domain during the close random pack
     domain.limits = [x + initial_radius - radius for x in domain.limits]
 
-    # Generate non-overlapping particles for an initial inner radius using
+    # Generate non-overlapping spheres for an initial inner radius using
     # random sequential packing algorithm
-    particles = _random_sequential_pack(domain, n_particles)
+    spheres = _random_sequential_pack(domain, num_spheres)
 
-    # Use the particle configuration produced in random sequential packing as a
-    # starting point for close random pack with the desired final particle
+    # Use the sphere configuration produced in random sequential packing as a
+    # starting point for close random pack with the desired final sphere
     # radius
-    if initial_packing_fraction != packing_fraction:
-        domain.particle_radius = radius
-        _close_random_pack(domain, particles, contraction_rate)
-
-    trisos = []
-    for p in particles:
-        trisos.append(TRISO(radius, fill, [x + c for x, c in zip(p, domain.center)]))
-    return trisos
+    if initial_pf != pf:
+        domain.sphere_radius = radius
+        _close_random_pack(domain, spheres, contraction_rate)
+
+    return spheres + domain.center
```

Comment:
Why `+ domain.center`?

Reply:
The methods for placing spheres assumed the container was centered on the origin -- I updated this in a commit I just pushed.

Reply:
Ah, got it -- some nice broadcasting going on there :+1: